tests/test_cli.py

```diff
@@ -1240,79 +1240,105 @@ def test_simulate_long_args(self):
+            )
+
+    def test_mutate_model_is_changeable(self):
+        parser = cli.get_msp_parser()
+        cmd = "mutate"
+        for model_arg in ["--model", "-m"]:
+            args = parser.parse_args(
+                [cmd, "out.trees", "out2.trees", model_arg, "blosum62"]
+            )
+            assert args.model == "blosum62"
+
 
-class TestMspSimulateOutput(unittest.TestCase):
+class TestMspSimulateOutput:
     """
     Tests the output of msp to ensure it's correct.
     """
 
-    def setUp(self):
-        fd, self._tree_sequence = tempfile.mkstemp(prefix="msp_cli", suffix=".trees")
-        os.close(fd)
-
-    def tearDown(self):
-        os.unlink(self._tree_sequence)
-
-    def test_run_defaults(self):
+    def test_run_defaults(self, tmp_path):
         cmd = "simulate"
         sample_size = 10
+        tree_sequence_file = str(tmp_path / "out.ts")
         stdout, stderr = capture_output(
-            cli.msp_main, [cmd, str(sample_size), self._tree_sequence]
+            cli.msp_main, [cmd, str(sample_size), tree_sequence_file]
         )
         assert len(stderr) == 0
         assert len(stdout) == 0
 
-        tree_sequence = tskit.load(self._tree_sequence)
+        tree_sequence = tskit.load(tree_sequence_file)
         assert tree_sequence.get_sample_size() == sample_size
         assert tree_sequence.get_sequence_length() == 1
         assert tree_sequence.get_num_mutations() == 0
 
-    def test_simulate_short_args(self):
+    def test_simulate_short_args(self, tmp_path):
         cmd = "simulate"
+        tree_sequence_file = str(tmp_path / "out.ts")
         stdout, stderr = capture_output(
             cli.msp_main,
-            [cmd, "100", self._tree_sequence, "-L", "1e2", "-r", "5", "-u", "2"],
+            [cmd, "100", tree_sequence_file, "-L", "1e2", "-r", "5", "-u", "2"],
         )
-        tree_sequence = tskit.load(self._tree_sequence)
+        tree_sequence = tskit.load(tree_sequence_file)
         assert len(stderr) == 0
         assert len(stdout) == 0
         assert tree_sequence.get_sample_size() == 100
         assert tree_sequence.get_sequence_length() == 100
         assert tree_sequence.get_num_mutations() > 0
 
-    def test_compress_warns(self):
+    def test_compress_warns(self, tmp_path):
         cmd = "simulate"
+        tree_sequence_file = str(tmp_path / "out.ts")
         with pytest.warns(UserWarning):
-            capture_output(cli.msp_main, [cmd, "10", self._tree_sequence, "--compress"])
-        tree_sequence = tskit.load(self._tree_sequence)
+            capture_output(cli.msp_main, [cmd, "10", tree_sequence_file, "--compress"])
+        tree_sequence = tskit.load(tree_sequence_file)
         assert tree_sequence.get_sample_size() == 10
 
 
-class TestMspConversionOutput(unittest.TestCase):
-    """
-    Tests the output of msp to ensure it's correct.
-    """
+@pytest.fixture
+def unmutated_tree_sequence():
+    return msprime.simulate(
```

Comment:
might as well make this ``sim_ancestry``, since we're living in the future

Reply:
Is this by any chance related to also setting `discrete=True` by default? 

The doc string for `sim_mutations` honestly reads like the `keep=True` argument is broken: It might work, but it might not, and if it does not work you'll get an exception, but there won't be much you can do to make it work.

Reply:
I have undone a few changes that I made previously. Test fixtures with mutations now still have the mutations. Four tests fail with errors about adding mutations, but I have marked them as xfail. I think that makes more sense?

Reply:
Yep, perfect. Ready to merge after a squahs.